Code/Framework/AzToolsFramework/AzToolsFramework/Viewport/LocalViewBookmarkLoader.cpp

```diff
@@ -395,107 +517,127 @@ namespace AzToolsFramework
-        {
-            return AZStd::string();
-        }
+    void LocalViewBookmarkLoader::OverrideStreamReadFn(StreamReadFn streamReadFn)
+    {
+        m_streamReadFn = AZStd::move(streamReadFn);
+    }
 
-        auto* prefabSystemComponent = AZ::Interface<AzToolsFramework::Prefab::PrefabSystemComponentInterface>::Get();
-        AZ_Assert(
-            prefabSystemComponent != nullptr,
-            "Prefab System Component Interface could not be found. "
-            "It is a requirement for the LocalViewBookmarkLoader class. "
-            "Check that it is being correctly initialized.");
+    void LocalViewBookmarkLoader::OverrideFileExistsFn(FileExistsFn fileExistsFn)
+    {
+        m_fileExistsFn = AZStd::move(fileExistsFn);
+    }
 
-        const auto prefabTemplate = prefabSystemComponent->FindTemplate(rootPrefabTemplateId);
-        const AZStd::string prefabTemplateName = prefabTemplate->get().GetFilePath().Filename().Stem().Native();
+    static AZ::Vector3 RadiansToDegrees(const AZ::Vector3& radians)
+    {
+        return AZ::Vector3(AZ::RadToDeg(radians.GetX()), AZ::RadToDeg(radians.GetY()), AZ::RadToDeg(radians.GetZ()));
+    }
 
-        // to generate the file name in which we will store the view Bookmarks we use the name of the prefab + the timestamp
-        // e.g. LevelName_1639763579377.setreg
-        const AZStd::chrono::system_clock::time_point now = AZStd::chrono::system_clock::now();
-        const AZStd::string timeSinceEpoch = AZStd::string::format("%llu", now.time_since_epoch().count());
-        return prefabTemplateName.data() + AZStd::string("_") + timeSinceEpoch + ".setreg";
+    static ViewBookmark ViewBookmarkFromCameraState(const AzFramework::CameraState& cameraState)
+    {
+        return { cameraState.m_position,
+                 RadiansToDegrees(AzFramework::EulerAngles(
+                     AZ::Matrix3x3::CreateFromColumns(cameraState.m_side, cameraState.m_forward, cameraState.m_up))) };
     }
 
-    void LocalViewBookmarkLoader::SetupLocalViewBookmarkComponent()
+    AZStd::optional<ViewBookmark> StoreViewBookmarkLastKnownLocationFromActiveCamera()
     {
-        LocalViewBookmarkComponent* bookmarkComponent = RetrieveLocalViewBookmarkComponent();
-        if (!bookmarkComponent)
+        bool found = false;
+        AzFramework::CameraState cameraState;
+        Camera::EditorCameraRequestBus::BroadcastResult(found, &Camera::EditorCameraRequestBus::Events::GetActiveCameraState, cameraState);
+
+        if (found)
         {
-            AZ_Warning("LocalViewBookmarkLoader", false, "Couldn't find a LocalViewBookmarkComponent");
-            return;
+            return StoreViewBookmarkLastKnownLocationFromCameraState(cameraState);
         }
 
-        // record an undo step if a local view bookmark component is added and configured
-        ScopedUndoBatch undoBatch("SetupLocalViewBookmarks");
-        undoBatch.MarkEntityDirty(bookmarkComponent->GetEntityId());
+        return {};
```

Comment:
minor: It'd be more explicit to use `AZStd::nullopt` when wanting to return a null optional. Same in the other functions below.

Reply:
Yeah good shout 👍